src/pytorch_lightning/core/datamodule.py

```diff
@@ -100,6 +114,15 @@ def get_init_arguments_and_types(cls) -> List[Tuple[str, Tuple, Any]]:
         """
         return get_init_arguments_and_types(cls)
 
+    @classmethod
+    def get_deprecated_arg_names(cls) -> List:
```

Comment:
Why is this added?

Reply:
mypy raises a union-attr error, as it is necessary in [`utilities.argparse.add_argparse_args`](https://github.com/JustinGoheen/lightning/blob/d2c086b04ad14ec5d6cf63e9487381a96eee1733/src/pytorch_lightning/utilities/argparse.py#L167); however, `get_deprecated_arg_names` isn't a function in `utilities.argparse`, it is a [class method of Trainer](https://github.com/JustinGoheen/lightning/blob/d2c086b04ad14ec5d6cf63e9487381a96eee1733/src/pytorch_lightning/trainer/trainer.py#L2433). 

I think the more correct thing to do, is to move the implementation of `get_deprecated_arg_names` to `utilities.argparse`, and then update `core.datamodule` and `trainer.trainer` accordingly ... do you agree?

Reply:
I agree. But let's do that in a PR before this one or a PR after.

Reply:
Sounds good to me. I'll open a new issue referring to this thread, and will follow up with a new PR after this one is merged.